alby/alby_oauth_service.go

```diff
@@ -1051,6 +1051,42 @@ func (svc *albyOAuthService) GetChannelPeerSuggestions(ctx context.Context) ([]C
 	return suggestions, nil
 }
 
+func (svc *albyOAuthService) GetBitcoinRate(ctx context.Context, currency string) (*BitcoinRate, error) {
+	client := &http.Client{Timeout: 10 * time.Second}
+
+	url := fmt.Sprintf("%s/rates/%s", albyInternalAPIURL, currency)
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		logger.Logger.WithError(err).Error("Error creating request to Bitcoin rate endpoint")
+		return nil, err
+	}
+	setDefaultRequestHeaders(req)
+
+	res, err := client.Do(req)
+	if err != nil {
+		logger.Logger.WithError(err).Error("Failed to fetch Bitcoin rate from API")
+		return nil, err
+	}
+
+	defer res.Body.Close()
+
+	if res.StatusCode != http.StatusOK {
```

Comment:
can you please check the others API calls to see what they do? I think this is a fragile check. It could be another 2XX or 3XX code (e.g. if the result was not modified)